src/gui/macutilities.mm

```diff
@@ -106,7 +106,15 @@ void openFiles(const PathList &pathList)
             for (const auto &path : pathList)
                 [pathURLs addObject:[NSURL fileURLWithPath:path.toString().toNSString()]];
 
-            [[NSWorkspace sharedWorkspace] activateFileViewerSelectingURLs:pathURLs];
```

Comment:
Since this is a Qt related issue maybe try to solve it without touching macOS related code (which is still correct)? I mean something like invoking openFiles using queued invocation?

Reply:
initially I did. but openFiles() is used in multiple places, so decided to add workaround exactly inside that function